__tests__/secret.test.js

```diff
@@ -110,4 +110,26 @@ describe('Tozny identity client', () => {
     expect(result[0].data.secretValue).toBe('secret-value')
     expect(result[0].meta.plain.secretType).toBe('Credential')
   })
+  it('can create a secret and update', async () => {
+    const secret = {
+      secretType: 'Credential',
+      secretName: `test-secret-updatetest35558800`,
+      secretValue: 'secret-value',
+      description: 'this is a description',
+    }
+    await ops.createSecret(realmConfig, identity, secret)
+    await new Promise(r => setTimeout(r, 1000))
+    const query = await ops.getSecrets(realmConfig, identity, 10)
+    let secrets = await query.next()
+    await ops.updateSecret(realmConfig, identity, secret)
+    await new Promise(r => setTimeout(r, 10000))
+    const query2 = await ops.getSecrets(realmConfig, identity, 100)
+    const newSecrets = await query2.next()
+    let length1 = secrets.length
+    let length2 = newSecrets.length
+    expect(length1 + 1).toBe(length2) // this shows we created a new secret
+    expect(secrets[length1 - 1].meta.plain.version).toBe('latest') // this shows that the last item is latest
+    expect(newSecrets[length1 - 1].meta.plain.version).toBe('v1') // this shows that now that item changed to V1
+    expect(newSecrets[length2 - 1].meta.plain.version).toBe('latest') // and the newest item is now latest
```

Comment:
Alright, this makes it clear I told you the wrong thing for how to version secrets, so lets try version 2 of secrets versioning

Every secret has a tag “version”, and a tag “version_number”. 
When the secret is first created, version = latest, version_number=1

When that secret is updated
Old secret version is updated to be archived, version number is unchanged
New secret is tagged with version=latest, version_number=2

Reply:
I think having a true latest tag that moves adds a lot of state and complexity that we don't want to manage, records also have version uuids so if someone were to be verifying a record based on this that would become out of date (I don't know if anyone would do that but it is something to think about). The original python version of secrets does use numbers for versions, but we have dates for when records are created, I don't really see why a counting versioning system is better than just a date based one.

Reply:
I may be trying to pre-maturely optimize something that isn't a problem, but I want a clear and fast way to say "get me the latest version of the secret" without having to fetch all the records, sort them by date, and take the newest one. So as long as we can keep the interface that developers or users want - a single method to get a secret that always returns the latest version, I'm fine with the date based approach or the "latest" tag way I originally proposed.

Reply:
100% agree on needing a simple way to ask that question. I would argue record type + write date limit 1 gets you there

Reply:
@rosasck talk with Eli about this and then implement the approach you two agree on please